### ISSUE TITLE
ignored-modules not ignored

### ISSUE THREAD

**Issue by crbunney** [PyCQA/pylint#3135]
In trying to work around https://github.com/PyCQA/astroid/issues/697, I added boto3 and botocore to a pylintrc file as `ignored-modules` but the error persisted.

Is this the correct way to disable checks relating to libraries with dynamically generate members?

### Steps to reproduce
1. Create a `pylint_test.py`:
```python
"""module docstring"""
import boto3
boto3.resource('sns').Topic('foo')
```
2. create a pylintrc file containing:
```
[TYPECHECK]
ignored-modules=boto3,botocore
```
3. pylint --rcfile=pylintrc pytlint_test.py

### Current behavior
```
$ pylint --rcfile=../pylintrc pytlint_test.py
************* Module pytlint_test
pytlint_test.py:3:0: E1101: Instance of '' has no 'Topic' member (no-member)
```

### Expected behavior
No errors reported

### pylint --version output
```
$ pylint --version
pylint 2.4.1
astroid 2.3.0
Python 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 22:22:05) [MSC v.1916 64 bit (AMD64)]
```

**Comment by PCManticore** [comment]
This should work but looks like a bug creeped, will investigate as part of 2.4.2.

**Comment by crbunney** [comment]
👍
